src/main/keep-container/keep-content/keep-add-notes/keep-add-notes.component.ts

```diff
@@ -42,6 +55,12 @@ export class KeepAddNotesComponent {
       this.showDropdownMenu = false;
       this.showFirstForm = true;
     }
+    if (this.showSecondForm && (this.elementRef.nativeElement.contains(event.target) || (this.FormElement && this.FormElement.nativeElement.contains(event.target)) || (this.Text && this.Text.nativeElement.contains(event.target))) && (this.secondFormElement && !this.secondFormElement.nativeElement.contains(event.target))) {
```

Comment:
can you please refactor this condition which will make it more readable and also please add comments to explain the logic